### PR TITLE
SDK-2230: Add example for share v2

### PR DESCRIPTION
**Pull request by irotech**

### CODE REVIEW

**Review by pn-santos — APPROVED** [review]
Just nits and one question so I don't want to block it.

File: yoti-sdk-spring-boot-example/src/main/java/com/yoti/api/examples/springboot/attribute/AttributeDisplayProperty.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    NATIONALITY("nationality", "Nationality", "yoti-icon-nationality"),
+    PHONE_NUMBER("phone_number", "Mobile number", "yoti-icon-phone"),
+    SELFIE("selfie"),
+    EMAIL_ADDRESS("email_address", "Email address", "yoti-icon-email"),
+    IDENTITY_PROFILE_REPORT("identity_profile_report", "Identity Profile Report", "yoti-icon-document", true);
+
+    final String name;
+    String display;
+    String icon;
+    boolean isJson;
```
Comment (pn-santos) [inline yoti-sdk-spring-boot-example/src/main/java/com/yoti/api/examples/springboot/attribute/AttributeDisplayProperty.java:26]: Shouldn't all these be `private final`?

File: yoti-sdk-spring-boot-example/README.md [hunk 2]
```diff
@@ -20,23 +22,28 @@ Before you start, you'll need to create an Application in [Yoti Hub](https://hub
...
+## Building your server-app and run the example
+* Copy the [resources/application.yml.example](src/main/resources/application.yml.example) and create a new file `resources/application.yml
+* Edit the `resources/application.yml` and replace the `yoti-client-sdk-id-from-hub` value with the `Yoti client SDK ID` you can find in Yoti Hub
+* Download your Application's key pair from Yoti Hub and copy it to `resources/app-keypair.pem`
+* Execute `mvn clean package` to build the project
 
-## Running
+### Share v1
+* In the Hub, set the scenario callback URL to `/login`. In order to receive calls on your /login endpoint, you need to expose your server-app to the outside world. We require that you use the domain from the Callback URL and HTTPS
 * You can run your server-app by executing `java -jar target/yoti-sdk-spring-boot-example-3.8.0-SNAPSHOT.jar`
```
[the remark this answers is not in the document]
Comment (irotech) [reply yoti-sdk-spring-boot-example/README.md:33]: I went for the 2nd option as it's cleaner

File: yoti-sdk-spring-boot-example/src/main/java/com/yoti/api/examples/springboot/attribute/AttributeMapper.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            case AttributeConstants.HumanProfileAttributes.EMAIL_ADDRESS:
+                return new DisplayAttribute("Email address", attribute, "yoti-icon-email");
+            case AttributeConstants.HumanProfileAttributes.DATE_OF_BIRTH:
+                return new DisplayAttribute("Date of birth", attribute, "yoti-icon-calendar");
+            case AttributeConstants.HumanProfileAttributes.SELFIE:
+                return null;
+            case AttributeConstants.HumanProfileAttributes.GENDER:
+                return new DisplayAttribute("Gender", attribute, "yoti-icon-gender");
+            case AttributeConstants.HumanProfileAttributes.IDENTITY_PROFILE_REPORT:
+                return new DisplayAttribute("Identity Profile Report", toJsonAttribute(attribute), "yoti-icon-document");
```
[the remark this answers is not in the document]
Comment (irotech) [reply yoti-sdk-spring-boot-example/src/main/java/com/yoti/api/examples/springboot/attribute/AttributeMapper.java]: I already changed this with a much clearer, at least for me, enum

File: yoti-sdk-spring-boot-example/src/main/java/com/yoti/api/examples/springboot/attribute/AttributeMapper.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        }
+    }
+
+    private static Attribute<String> toJsonAttribute(Attribute<?> attribute) {
+        String json;
+
+        try {
+            json = MAPPER.readTree(
+                    MAPPER.writeValueAsString(attribute.getValue()).getBytes(StandardCharsets.UTF_8)
+            ).toString();
```
[the remark this answers is not in the document]
Comment (irotech) [reply yoti-sdk-spring-boot-example/src/main/java/com/yoti/api/examples/springboot/attribute/AttributeMapper.java]: Yeah I noticed while reviewing as I just moved this legacy code. Already fixed.

File: yoti-sdk-spring-boot-example/src/main/resources/templates/error.html [hunk 5]
```diff
@@ -1,14 +1,27 @@
...
-		<title>Welcome</title>
-		<link rel="stylesheet" type="text/css" th:href="@{/static/style.css}" />
-		<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
-	</head>
-
-	<body>
-		<h2><a href="/">Home</a></h2>
-		<h3>Could not login user for the following reason: <span th:text="${error} "/> </h3>
-	</body>
+<!DOCTYPE html>
```
[the remark this answers is not in the document]
Comment (irotech) [reply yoti-sdk-spring-boot-example/src/main/resources/templates/error.html:1]: Read a bit about it and u're right Thymeleaf processes the template on the server side and includes this tag in the resulting HTML for the browser to render.

File: yoti-sdk-spring-boot-example/src/main/resources/templates/digital-identity-share.html [hunk 6]
```diff
@@ -1,56 +1,80 @@
...
-            <p> Receipt ID: <strong th:text="${fetch_session_receipt_id}"></strong></p>
-        </div>
-    </section>
-</main>
-
-</body>
+
+    <head>
+        <meta charset="UTF-8">
+        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"/>
```
[the remark this answers is not in the document]
Comment (irotech) [reply yoti-sdk-spring-boot-example/src/main/resources/templates/digital-identity-share.html:7]: We make the standars!!! ^_^

Changed indent to 2 .. quite compact now

File: yoti-sdk-spring-boot-example/README.md [hunk 7]
```diff
@@ -1,16 +1,17 @@
...
-This project shows an example implementation of a server-app with an endpoint which will be called by Yoti with a `token`.
-You will need to pass this token to Yoti-SDK in order to retrieve the profile of a user which has been logged in by Yoti.
+# Prerequisites
+Before you start, you'll need to create an Application in [Yoti Hub](https://hub.yoti.com) and set the domain to `https://localhost:8443/`. 
 
-Before you start, you'll need to create an Application in [Yoti Hub](https://hub.yoti.com) and verify the domain.
-
-**NOTE: While creating Application in Yoti Hub, some of the attributes (except phone number and selfie) require users to have a Yoti with a verified passport. If your application, for instance, requires the user's date of birth and she/he has not added their passport to their Yoti account, this will lead to a failed login.**
+Note that:
+- Your endpoint must be accessible from the machine that is displaying the QR code.**
```
Comment (pn-santos) [inline yoti-sdk-spring-boot-example/README.md:8]: the `**` at the end are leftovers should be removed

File: yoti-sdk-spring-boot-example/src/main/java/com/yoti/api/examples/springboot/IdentitySessionController.java [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+        this.client = client;
+    }
+
+    @GetMapping("/digital-identity-session")
+    public String identityShareSession() {
+        ShareSession session = client.createShareSession(
+                // forMinimalShare()
+                forDynamicScenarioShare()
+                // forIdentityProfileShare()
+                // forLocationExtensionShare()
```
Comment (pn-santos) [inline yoti-sdk-spring-boot-example/src/main/java/com/yoti/api/examples/springboot/IdentitySessionController.java:42]: Are these commented out on purpose? if so then maybe leave a comment explaining why they are commented out? or how/when should the commented out parts be un-commented?

**Review by pn-santos — APPROVED** [review]

File: yoti-sdk-spring-boot-example/src/main/java/com/yoti/api/examples/springboot/IdentitySessionController.java (hunk 8 above)
Comment (irotech) [reply yoti-sdk-spring-boot-example/src/main/java/com/yoti/api/examples/springboot/IdentitySessionController.java:42]: yes, I created a quick to switch scenarios for Darren. If he wants to create more tests he is free to do so

**Pull request merged**